src/renderer/api/catalog-entity-registry.ts

```diff
@@ -27,15 +27,25 @@ import type { Cluster } from "../../main/cluster";
 import { ClusterStore } from "../../common/cluster-store";
 import { Disposer, iter } from "../utils";
 import { once } from "lodash";
+import type { CatalogEntityItem } from "../../renderer/components/+catalog/catalog-entity-item";
 
 export type EntityFilter = (entity: CatalogEntity) => any;
+export type CatelogEntityOnRunHook = (entity: CatalogEntity | CatalogEntityItem<CatalogEntity>) => boolean | Promise<boolean>;
+
+type CatalogEntityUid = CatalogEntity["metadata"]["uid"];
```

Comment:
I would just use `string` directly.

Reply:
IMHO this is more descriptive we are using uid from `catalog.metadata.uid`

Reply:
Ok